foster55f/market2table-fe
# Vendor image upload

### What does this PR do?
- Implements upload photo functionality
- installs react-dropzone and superagent and integrates with Cloudinary to host images
- react-dropzone gives us access to components and methods to apply a dropzone to any elements. Will only accept certain file types and only one image can be uploaded.
- superagent allows us to post and send image to Cloudinary
- Cloudinary hosts the image and sends us back a url that can be used to display image and stored in the database
- displays image on form once it is uploaded
- ability to delete image on click
- Adds media queries to the form to make it responsive based on screen size.
- Adds error handling to require vendorname, image and description in order to create a new one.

### How to test?
 - Pull down branch
 - run 'npm start' and the site should open in browser via localhost:3000
 - run 'npm install'
- Click Create New Vendor button and then either click dropzone or drag and drop a file and see functionality. Try deleting it.

### Issues:
 - The POST for adding products still needs to be implemented on the same submit vendor button click. 
- Need to have the form work for editing a page as well as creating.
- Error handling for products being submitted.

### Notes:
- https://github.com/react-dropzone/react-dropzone
- https://react-dropzone.js.org/#basic%20example
- https://cloudinary.com/
- https://www.npmjs.com/package/superagent

### Relevant Screenshots:

<img width="1427" alt="Screen Shot 2020-02-23 at 11 41 52 AM" src="https://user-images.githubusercontent.com/49846853/75117810-307be180-5632-11ea-8ae3-87042c5eee3d.png">

<img width="1427" alt="Screen Shot 2020-02-23 at 11 42 20 AM" src="https://user-images.githubusercontent.com/49846853/75117811-3376d200-5632-11ea-8444-d8ea66a55f43.png">
